### PR TITLE
[CI] New workflows order part1

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
Please review each commit separately. Verified on my fork it all works as expected.

Notes:
- Unfortunately currently GHA does not support sub-dirs in `.github/workflows` directory.
- Scorecard is not giving a penalty now for reusable workflows with higher permissions.

### Checklist
- [x] CI workflows execute properly

### CODE REVIEW

**Review by PatKamin — APPROVED** [review]

**Review by PatKamin — APPROVED** [review]

**Comment by lukaszstolarczuk** [comment]
Code coverage did not change - still 80,1% - sounds like I didn't break anything with this change :)

File: .github/workflows/reusable_valgrind.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Run tests with valgrind intstrumentation tools: memcheck, drd, helgrind
```
Comment (ldorau) [inline .github/workflows/reusable_valgrind.yml:1]: `intstrumentation` -> `instrumentation`

File: .github/workflows/reusable_docs_build.yml [hunk 2]
```diff
@@ -21,11 +21,13 @@ jobs:
         sudo apt-get update
         sudo apt-get install -y doxygen
 
-    - name: Install pip requirements
-      run: python3 -m pip install -r third_party/requirements.txt
-
-    - name: Setup PATH for python
-      run: echo "$HOME/.local/bin" >> $GITHUB_PATH
```
Comment (ldorau) [inline .github/workflows/reusable_docs_build.yml]: Isn't it needed any more?

**Review by ldorau — APPROVED** [review]

File: .github/workflows/reusable_docs_build.yml (hunk 2 above)
Comment (lukaszstolarczuk) [reply .github/workflows/reusable_docs_build.yml]: nope, it's part of the `Install Python requirements in venv` step, pls see: `echo "$PATH" >> $GITHUB_PATH` (prev. it was installed in this `.local/bin` dir, now it's installed within venv)

File: .github/workflows/reusable_valgrind.yml (hunk 1 above)
Comment (lukaszstolarczuk) [reply .github/workflows/reusable_valgrind.yml:1]: I'll fix that in the next PR; I'd like to omit waiting for the whole CI to re-run 😄

**Pull request merged**